sno/branch.py

```diff
@@ -0,0 +1,86 @@
+import json
+import sys
+
+import click
+import pygit2
+
+from .cli_util import MutexOption
+from .exec import execvp
+
+
+@click.command(context_settings=dict(ignore_unknown_options=True))
+@click.pass_context
+@click.option(
```

Comment:
Could we make this a single decorator we re-use?

eg:
```python
def sno_output_format_options(func):
  """ Apply Text & JSON output format options to a Click command """
  func = click.option("--text", ...)(func)
  func = click.option("--json", ...)(func)
  return func

@click.command(context_settings=dict(ignore_unknown_options=True))
@click.pass_context
@sno_output_format_options
@click.argument("args", nargs=-1, type=click.UNPROCESSED)
def branch(ctx, is_output_json, args):
  ...
```

Reply:
Will do this in a separate PR